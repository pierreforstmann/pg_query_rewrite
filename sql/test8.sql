drop extension if exists pg_query_rewrite;
--
create extension pg_query_rewrite;
select pgqr_truncate();
--
select pgqr_add_rule('select 10;','select 11;');
select pgqr_add_rule('select ''Hello'';','select ''Good Bye'';');
--
select 10;
select 100;
select 'Hello';
select 'Hello ...';
--
select pgqr_add_rule('select ''Hello'';','select ''Good Bye'';');
select pgqr_add_rule('select 10;','select 11;');
--
select 10;
select 100;
select 'Hello';
select 'Hello ...';
--
select pgqr_remove_rule('select 10;');
select 10;
select 'Hello';
select pgqr_remove_rule('select ''Hello'';');
select 'Hello';
--
select pgqr_rules();
--
drop extension pg_query_rewrite;
