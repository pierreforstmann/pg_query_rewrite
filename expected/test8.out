drop extension if exists pg_query_rewrite;
--
create extension pg_query_rewrite;
select pgqr_truncate();
 pgqr_truncate 
---------------
 t
(1 row)

--
select pgqr_add_rule('select 10;','select 11;');
 pgqr_add_rule 
---------------
 t
(1 row)

select pgqr_add_rule('select ''Hello'';','select ''Good Bye'';');
 pgqr_add_rule 
---------------
 t
(1 row)

--
select 10;
 ?column? 
----------
       11
(1 row)

select 100;
 ?column? 
----------
      100
(1 row)

select 'Hello';
 ?column? 
----------
 Good Bye
(1 row)

select 'Hello ...';
 ?column?  
-----------
 Hello ...
(1 row)

--
select pgqr_add_rule('select ''Hello'';','select ''Good Bye'';');
ERROR:  rule already exists for select 'Hello';
select pgqr_add_rule('select 10;','select 11;');
ERROR:  rule already exists for select 10;
--
select 10;
 ?column? 
----------
       11
(1 row)

select 100;
 ?column? 
----------
      100
(1 row)

select 'Hello';
 ?column? 
----------
 Good Bye
(1 row)

select 'Hello ...';
 ?column?  
-----------
 Hello ...
(1 row)

--
select pgqr_remove_rule('select 10;');
 pgqr_remove_rule 
------------------
 t
(1 row)

select 10;
 ?column? 
----------
       10
(1 row)

select 'Hello';
 ?column? 
----------
 Good Bye
(1 row)

select pgqr_remove_rule('select ''Hello'';');
 pgqr_remove_rule 
------------------
 t
(1 row)

select 'Hello';
 ?column? 
----------
 Hello
(1 row)

--
select pgqr_rules();
                       pgqr_rules                       
--------------------------------------------------------
 (datname=NULL,source=NULL,target=NULL,rewrite_count=0)
 (datname=NULL,source=NULL,target=NULL,rewrite_count=0)
 (datname=NULL,source=NULL,target=NULL,rewrite_count=0)
 (datname=NULL,source=NULL,target=NULL,rewrite_count=0)
 (datname=NULL,source=NULL,target=NULL,rewrite_count=0)
 (datname=NULL,source=NULL,target=NULL,rewrite_count=0)
 (datname=NULL,source=NULL,target=NULL,rewrite_count=0)
 (datname=NULL,source=NULL,target=NULL,rewrite_count=0)
 (datname=NULL,source=NULL,target=NULL,rewrite_count=0)
 (datname=NULL,source=NULL,target=NULL,rewrite_count=0)
(10 rows)

--
drop extension pg_query_rewrite;
